drivers/power/adm1177/adm1177.c

```diff
@@ -0,0 +1,311 @@
+{
+	return no_os_i2c_write(device->i2c_desc, &command_byte, 1, 1);
+}
+
+/***************************************************************************//**
+ * @brief - The quick command operation allows the master to check if the slave
+ *          is present on the bus.
+ *
+ * @param dev - The device structure.
+ *
+ * @param slave_address - The slave address, by default is 0xA5.
+ *
+ * @param reg_address - MSB set to 1, indicates an extended register write. The
+ *                      two LSBs indicate which of the three extended registers
+ *                      is to be written to. All other bits should be set to 0.
+ *
+ * @param ext_cmd_byte - The extended command byte indicates which operation of
+ *                       the specified register should be used. Each register
+ *                       has its own operations.
+ *
+ * @return 0 in case of success, -1 otherwise.
+*******************************************************************************/
+int32_t adm1177_write_extended_command_byte(struct adm1177_dev *device,
+		uint8_t reg_address,
+		uint8_t ext_cmd_byte)
+{
+	uint8_t data[2] = {reg_address, ext_cmd_byte};
+
+	return no_os_i2c_write(device->i2c_desc, data, 2, 1);
+}
+
+/***************************************************************************//**
+ * @brief - The read status function reads the status byte which has its own
+ *          operations. Status Byte is read only when C6 in command byte is 1.
+ *
+ * @param dev - The device structure.
+ *
+ * @param slave_address - The slave address, by default is 0xA5.
+ *
+ * @param status_byte - The status byte which is received.
+ *
+ * @return 0 in case of success, -1 otherwise.
+*******************************************************************************/
+int32_t adm1177_read_status(struct adm1177_dev *device,
+			    uint8_t *status_byte)
+{
+	int32_t ret;
+
+	ret = no_os_i2c_read(device->i2c_desc, status_byte, 1, 1);
+
+	if(ret)
+		return -1;
+
+	return 0;
+}
+
+/***************************************************************************//**
+ * @brief - Gets the voltage/current readback from the ADM1177s I2C ADC,
+ *          if the device is initialized right and transmits the requested
+ *          values.
+ *
+ * @param dev - The device structure.
+ *
+ * @param conv_voltage - The variable through which the voltage read from the ADC
+ *                       is received if requested.
+ *
+ * @param conv_current - The variable through which the current read from the ADC
+ *                       is received if requested.
+ *
+ * @return 0 in case of success, -1 otherwise.
+*******************************************************************************/
+int32_t adm1177_readback(struct adm1177_dev *device,
+			 uint16_t* conv_voltage,
+			 uint16_t* conv_current)
+{
+	int32_t ret;
+
+	uint8_t data[3];
+
+	ret = no_os_i2c_read(device->i2c_desc, data, 3, 1);
+
+	if(ret)
+		return -1;
+
+	*conv_voltage = (data[0] << 4) | (data[2] >> 4);
```

Comment:
would be nice to either add some comments explaining what you do here with respect to the data received **or** use `no_os_field_get` in conjunction with relevant masks.